### PR TITLE
Added CUUID to Glibc

### PR DESCRIPTION
**Pull request by colemancda**
#### What's in this pull request?

Adds libUUID to Glibc so the UUID library can be imported without third party modules like it currently is on Darwin (e.g. `import Darwin.uuid`)
#### Resolved bug number: ([SR-1756](https://bugs.swift.org/browse/SR-1756))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by hpux735** [comment]
Replaces https://github.com/apple/swift/pull/3095

**Comment by phausler** [comment]
Overall looks good to me, this will definitely help the story for struct UUID in swift-corelibs-foundation

**Comment by phausler** [comment]
@swift-ci Please test

File: validation-test/stdlib/CUUID.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// RUN: %target-run-stdlib-swift
+// REQUIRES: executable_test
+
+// REQUIRES: OS=linux-gnu
+
+import Swift
```
Comment (gribozavr) [inline validation-test/stdlib/CUUID.swift]: You don't need to `import Swift` if you use `%target-run-simple-swift`.

File: validation-test/stdlib/CUUID.swift [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import StdlibUnittest
+
+
```
Comment (gribozavr) [inline validation-test/stdlib/CUUID.swift]: Please remove the double empty lines.

File: validation-test/stdlib/CUUID.swift [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+var CUUIDTestSuite = TestSuite("CUUID")
+
+CUUIDTestSuite.test("uuid") {
+  var uuid: uuid_t = (0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)
+  withUnsafeMutablePointer(&uuid) {
+    uuid_generate_random(unsafeBitCast($0, to: UnsafeMutablePointer<UInt8>.self))
+  }
+}
+
+runAllTests()
```
Comment (gribozavr) [inline validation-test/stdlib/CUUID.swift]: Please add a newline.

File: validation-test/stdlib/CUUID.swift [hunk 4]
```diff
@@ -0,0 +1,21 @@
+// RUN: %target-run-stdlib-swift
+// REQUIRES: executable_test
+
+// REQUIRES: OS=linux-gnu
+
+import Swift
+import StdlibUnittest
+
+
+import CUUID
```
Comment (gribozavr) [inline validation-test/stdlib/CUUID.swift]: Could you add a comment to the test that explains what it does?  "Make sure that the module map for the uuid.h header works."

**Comment by colemancda** [comment]
@gribozavr Applied changes.

**Comment by gribozavr** [comment]
@swift-ci Please test and merge

**Pull request merged**